File: Java OOP Concept/Polymorphism.java
class Animal {
    void makeSound() {
        System.out.println("Some generic animal sound");
    }
}

class Cat extends Animal {
    @Override
    void makeSound() {
        System.out.println("Meow!");
    }
}

public class Polymorphism {
    public static void main(String[] args) {
        Animal myAnimal = new Cat();
        myAnimal.makeSound();
    }
}

File: Flutter or Dart OOP Concept/Flutter OOP/Polymorphism.dart
class Animal {
	void makeSound() {
		print('Some generic animal sound');
	}
}

class Cat extends Animal {
	@override
	void makeSound() {
		print('Meow!');
	}
}

void main() {
	Animal myAnimal = Cat();
	myAnimal.makeSound();
}

